src/pager.rs

```diff
@@ -0,0 +1,120 @@
+use shell_words::ParseError;
+use std::env;
+
+/// If we use a pager, this enum tells us from where we were told to use it.
+#[derive(Debug, PartialEq)]
+pub enum PagerSource {
+    /// From --config
+    Config,
+
+    /// From the env var BAT_PAGER
+    BatPagerEnvVar,
+
+    /// From the env var PAGER
+    PagerEnvVar,
+
+    /// No pager was specified, default is used
+    Default,
+}
+
+/// We know about some pagers, for example 'less'. This is a list of all pagers we know about
+#[derive(Debug, PartialEq)]
+pub enum PagerKind {
+    /// The pager is ourselves
+    Bat,
+
+    /// less
+    Less,
+
+    /// more
+    More,
+
+    /// most
+    Most,
+
+    /// A pager we don't know about
+    Unknown,
+}
+
+impl PagerKind {
+    fn from_bin(bin: &str) -> PagerKind {
+        use std::ffi::OsStr;
+        use std::path::Path;
+
+        let stem = Path::new(bin)
+            .file_stem()
+            .unwrap_or_else(|| OsStr::new("unknown"));
+
+        if stem == OsStr::new("bat") {
+            PagerKind::Bat
+        } else if stem == OsStr::new("less") {
+            PagerKind::Less
+        } else if stem == OsStr::new("more") {
+            PagerKind::More
+        } else if stem == OsStr::new("most") {
+            PagerKind::Most
+        } else {
+            PagerKind::Unknown
+        }
+    }
+}
+
+/// A pager such as 'less', and from where we got it.
+#[derive(Debug)]
+pub struct Pager {
+    // The pager binary
+    pub bin: String,
```

Comment:
I'll happily fix that, but prefer to do it in a separate follow-up PR, since this PR is not making it worse, so good that you think it's OK to do so.

I agree it's nicer to use types that better represents the data that you're working with.